Create anvilprod deployment and build on GitLab

hannes-ucsc opened the pull request:
Same as https://github.com/DataBiosphere/anvil-portal/pull/1 in UCSC's fork, but for the upstream repository.

bvizzier-ucsc commented:
@NoopDog Please review today.

NoopDog commented on src/config/anvil/config.ts:
```diff
@@ -26,7 +26,7 @@ const config: SiteConfig = {
         {
           label: "Datasets",
-          url: `${getDatasetsEnvironmentUrl()}data`,
+          url: "/explore/datasets",
```
@hannes-ucsc can you please revert this change? Thanks! D

hannes-ucsc replied:
I anticipated this change being controversial. I didn't understand `getDatasetsEnvironmentUrl` so I just clobbered it with the hard-coded line. Is there any way this could be made dynamic so that we don't have to maintain a branch for this one-line difference? We have a way to inject environment variables into the build. See my next comment.

NoopDog replied:
We are in the process of implementing the catalog/explorer selector now, so how about for right now, we make this PR deal with the prod build and deploy only, and we remove the setting of the value for the "Datasets" link as this will change soon anyhow.

We want to keep the catalog/explorer selector on its own branch until we are ready to deploy this to anvilproject.org, which I believe is in several weeks at the earliest. We will need to do deployments until the switchover, so I don't want to make changes to `main` that change the functionality for the managed access deployment only.

If I have the schedule wrong, let me know.
cc: @bvizzier-ucsc

NoopDog reviewed (changes requested):
Hi @hannes-ucsc I don't have any comment on the .gitlab section but have requested you revert the change to the header config.  I would like to suggest we update the config for https://prod.anvil.gi.ucsc.edu/ on a branch and not on main and merge into main once the site goes live on anvilproject.org.

hannes-ucsc commented on .gitlab/fragments/pipeline.yaml:
```diff
@@ -0,0 +1,3 @@
+variables:
```
This is where the environment variables are set.

NoopDog replied:
@hannes-ucsc, @bvizzier-ucsc You need to do the equivalent of `export GATSBY_ENV="ANVIL-PROD"` before running the build.

See:

https://github.com/anvilproject/anvil-portal/blob/107c9c0ed1cf000a102e9bd1bca728a842cf57c8/src/utils/environment/environment.service.js

And
https://github.com/anvilproject/anvil-portal/blob/107c9c0ed1cf000a102e9bd1bca728a842cf57c8/src/utils/environment/environment-url.model.js

Then you can remove the clobbered line.

Cheers,
Dave

hannes-ucsc replied:
I removed the commit.

hannes-ucsc replied:
We are setting GATSBY_ENV in base.yaml below.

> Then you can remove the clobbered line.

That surprises me since the [clobbered line](https://github.com/anvilproject/anvil-portal/blob/107c9c0ed1cf000a102e9bd1bca728a842cf57c8/src/config/anvil/config.ts#L29) always appends `data` which wouldn't work in this case.

hannes-ucsc replied:
But again, I removed the commit and will let you figure this out.

hannes-ucsc commented:
@NoopDog, I removed the problematic commit. I think you are in a better position to implement the equivalent functionality i.e., the backlink from portal to browser. I propose that you review this, and assuming you approve, merge it into the `main` branch, and then make the necessary changes for the back link from portal to browser.

We are also open to using a different branch to track the work on prod.anvil.gi.ucsc.edu as long as that branch is never force-pushed to.

@achave11-ucsc knows how to take changes from this repository and deploy them to prod.anvil.gi.ucsc.edu. Please let him know what the branch name is, and when that branch contains the equivalent to the commit I removed. He can then take the necessary steps to deploy that branch to prod.anvil.gi.ucsc.edu.

hannes-ucsc commented:
I added a commit that adds a build for the consortia site. You can preview this here:

https://anvil.gi.ucsc.edu/consortia

Not being an expert in this code base, I followed the same approach as before and did not bother with fixing any links in the top nav of the site. The deployment part is in place, the rest is configuration, which I'll leave to you.

hannes-ucsc commented:
The corresponding Azul PR is 

https://github.com/DataBiosphere/azul/pull/5453

NoopDog commented:
Not required
